mlir/lib/Dialect/PDL/IR/Builtins.cpp

```diff
@@ -39,6 +39,30 @@ mlir::Attribute addElemToArrayAttr(mlir::PatternRewriter &rewriter,
   values.push_back(element);
   return rewriter.getArrayAttr(values);
 }
+
+LogicalResult equals(mlir::PatternRewriter &, mlir::Attribute lhs,
+                     mlir::Attribute rhs) {
+  if (auto lhsAttr = dyn_cast_or_null<IntegerAttr>(lhs)) {
+    auto rhsAttr = dyn_cast_or_null<IntegerAttr>(rhs);
+    if (!rhsAttr || lhsAttr.getType() != rhsAttr.getType())
+      return failure();
+
+    APInt lhsVal = lhsAttr.getValue();
+    APInt rhsVal = rhsAttr.getValue();
+    return success(lhsVal.eq(rhsVal));
+  }
+
+  if (auto lhsAttr = dyn_cast_or_null<FloatAttr>(lhs)) {
+    auto rhsAttr = dyn_cast_or_null<FloatAttr>(rhs);
+    if (!rhsAttr || lhsAttr.getType() != rhsAttr.getType())
+      return failure();
+
+    APFloat lhsVal = lhsAttr.getValue();
+    APFloat rhsVal = rhsAttr.getValue();
+    return success(lhsVal.compare(rhsVal) == llvm::APFloatBase::cmpEqual);
+  }
+  return failure();
+}
```

Comment:
Why only IntegerAttr and floatAttr and why don't we use == operator from `Attribute` class?